src/box_test.jl

```diff
@@ -0,0 +1,87 @@
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+export BoxPierceTest, LjungBoxTest
+
+# Box-Pierce test
+
+immutable BoxPierceTest <: HypothesisTest
+    y::Vector{Float64}  # time series vector
+    n::Int              # number of observations
+    lag::Int            # number of lags in test statistic
+    dof::Int            # degrees-of-freedom correction (if y is residual)
+    Q::Float64          # test statistic
+end
+
+function BoxPierceTest{T<:Real}(y::AbstractVector{T}, lag::Int, dof::Int=0)
+    if dof>=lag
+        throw(ArgumentError("Number of lags has to be larger than d-o-f correction"))
+    end
+	n = size(y)[1]
+	Q = n * mapreduce(k -> autocor(y,k:k).^2, +, 1:lag)[1]
+	BoxPierceTest(y,n,lag,dof,Q)
+end
+
+testname(::BoxPierceTest) = "Box-Pierce autocorrelation test"
+population_param_of_interest(x::BoxPierceTest) = ("not applicable", NaN, NaN)
+
+function show_params(io::IO, x::BoxPierceTest, ident)
+    println(io, ident, "number of observations: ", x.n)
+    println(io, ident, "number of lags:         ", x.lag)
+    println(io, ident, "d-o-f correction:       ", x.dof)
+    println(io, ident, "Q statistic:            ", x.Q)
+end
+
+pvalue(x::BoxPierceTest) = pvalue(Chisq(x.lag-x.dof), x.Q; tail=:right)
+
+#Ljung-Box test
+
+immutable LjungBoxTest <: HypothesisTest
+    y::Vector{Float64}  # time series vector
+    n::Int              # number of observations
+    lag::Int            # number of lags in test statistic
+    dof::Int            # degrees-of-freedom correction (if y is residual)
+    Q::Float64          # test statistic
+end
+
+function LjungBoxTest{T<:Real}(y::AbstractVector{T}, lag::Int, dof::Int=0)
+    if dof>=lag
+        throw(ArgumentError("Number of lags has to be larger than d-o-f correction"))
+    end
+	n = size(y)[1]
+	Q = n*(n+2)* mapreduce(k -> autocor(y,k:k).^2/(n-k), +, 1:lag)[1]
+	LjungBoxTest(y,n,lag,dof,Q)
+end
+
+testname(::LjungBoxTest) = "Ljung-Box autocorrelation test"
+population_param_of_interest(x::LjungBoxTest) = ("not applicable", NaN, NaN)
```

Comment:
It might be helpful to think of a regression where you want to test a number of joint hypotheses using an F- (or Wald) statistic. You would not write H_0: F=0.